satpy/node.py

```diff
@@ -17,16 +17,22 @@
 # satpy.  If not, see <http://www.gnu.org/licenses/>.
 """Nodes to build trees."""
 
-from satpy.dataset import DataID, DataQuery, ModifierTuple
-from satpy.readers import TooManyResults, get_key
 from satpy.utils import get_logger
-from satpy.dataset import create_filtered_query
 
 LOG = get_logger(__name__)
 # Empty leaf used for marking composites with no prerequisites
 EMPTY_LEAF_NAME = "__EMPTY_LEAF_SENTINEL__"
 
 
+class MissingDependencies(RuntimeError):
+    """Exception when dependencies are missing."""
+
+    def __init__(self, missing_dependencies, *args, **kwargs):
+        """Set up the exception."""
+        super().__init__(*args, **kwargs)
+        self.missing_dependencies = missing_dependencies
```

Comment:
Can/should this have a custom __repr__ or __str__ to provide the typical missing dependency error message?

Reply:
Good idea

Reply:
Fixed